projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,603 @@
+            case KEYS.UP_ARROW:
+            case KEYS.UP_ARROW_IE:
+                if (event.altKey) {
+                    this.close();
+                }
+                break;
+            case KEYS.DOWN_ARROW:
+            case KEYS.DOWN_ARROW_IE:
+                if (event.altKey) {
+                    this.open();
+                }
+                break;
+            case KEYS.ESCAPE:
+            case KEYS.ESCAPE_IE:
+                this.close();
+                break;
+        }
+    }
+
+    /** @hidden @internal */
+    public handleSelection(selectionData: Date[]): void {
+        this.value = this.extractRange(selectionData);
+        this._onChangeCallback(this.value);
+        this.onSelected.emit(this.value);
+    }
+
+    /** @hidden @internal */
+    public selectToday(): void {
+        this.selectRange(new Date());
+    }
+
+    /** @hidden @internal */
+    public onClick(event: MouseEvent): void {
+        if (event) {
+            event.stopPropagation();
+            event.preventDefault();
+        }
+        this.open();
+    }
+
+    /** @hidden @internal */
+    public onCalendarOpened(): void {
+        requestAnimationFrame(() => {
+            this.calendar.daysView.focusActiveDate();
+        });
+    }
+
+    private updateCalendar() {
+        if (this.value && this.value.start && this.value.end) {
+            this.calendar.selectDate([this.value.start, this.value.end]);
+        } else {
+            if (this.value && this.value.start) {
+                this.calendar.selectDate(this.value.start);
+            } else if (this.value && this.value.end) {
+                this.calendar.selectDate(this.value.end);
+                this.value = { start: this.value.end, end: this.value.end };
+            } else {
+                this.calendar.deselectDate();
+            }
+        }
+        if (this.value) {
+            this.calendar.viewDate = this.value.start;
+        }
+    }
+
+    private extractRange(selection: Date[]): DateRange {
+        return {
+            start: selection[0],
+            end: selection.length > 1 ? selection[selection.length - 1] : null
+        };
+    }
+
+    private activateToggleOpen(overlaySettings: OverlaySettings): void {
+        if (this.toggleDirective.collapsed) {
+            this.toggleDirective.open(overlaySettings);
+        }
+    }
+
+    private attachOnKeydown(): void {
+        fromEvent(this.element.nativeElement, 'keydown')
+            .pipe(takeUntil(this._destroy))
+            .subscribe((evt: KeyboardEvent) => this.onKeyDown(evt));
+    }
+
+    private applyFocusOnClose() {
+        this.toggleDirective.onClosed
+            .pipe(takeUntil(this._destroy))
+            .subscribe(() => (
+                this.single || this.projectedInputs.find(i => i instanceof IgxDateStartComponent) as IgxDateStartComponent
+            )?.setFocus());
+    }
+
+    private subscribeToDateEditorEvents() {
+        if (this.hasProjectedInputs) {
```

Comment:
this checks if you have either `IgxDateStartComponent` or `IgxDateEndComponent`. You should check then if you have both of them if in some weird scenario you end with only start or only end